### PR TITLE
Add InsertTextFormat enum

### PR DESCRIPTION
**Pull request by muffinmad**
## Description (e.g. "Related to ...", etc.)

Add `InsertTextFormatEnum` for [CompletionItem](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion).
Also change type annotation and default value of `insert_text_format` to `int = 1`.

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [ ] [CONTRIBUTORS.md][contributors] was updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/pygls/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/
[contributors]: https://github.com/openlawlibrary/pygls/blob/master/CONTRIBUTORS.md

### CODE REVIEW

File: pygls/types.py [hunk 1]
```diff
@@ -293,6 +293,11 @@ def __init__(self, value_set: List['CompletionItemKind']):
         self.valueSet = value_set
 
 
+class InsertTextFormat(enum.IntEnum):
```
Comment (danixeee) [inline pygls/types.py:296]: Definition of `InsertTextFormat` class already exists [here](https://github.com/openlawlibrary/pygls/blob/master/pygls/types.py#L704).

File: pygls/types.py [hunk 2]
```diff
@@ -230,7 +230,7 @@ def __init__(self,
                  sort_text: str = None,
                  filter_text: str = None,
                  insert_text: str = None,
-                 insert_text_format: str = None,
+                 insert_text_format: int = 1,
```
Comment (danixeee) [inline pygls/types.py:233]: Could you please replace `1` with `InsertTextFormat. PlainText` as a default value?

File: pygls/types.py (hunk 1 above)
Comment (muffinmad) [reply pygls/types.py:296]: Totally missed this. Fixed.

File: pygls/types.py (hunk 2 above)
Comment (muffinmad) [reply pygls/types.py]: How about to change just annotation type and leave default value unspecified? Please see updated patch.

Comment (danixeee) [reply pygls/types.py]: Sounds good.

**Pull request merged**